### ISSUE TITLE
EV3/UART sensors detected but don't appear under /sys/class/lego-sensor

### ISSUE THREAD

**Issue by jfcloutier** [nerves-project/nerves_system_ev3#26]
UART sensors are detected but do not appear under /sys/class/lego-sensor.

```
iex(andy@nerves-5e58)58> ls "/sys/class/lego-sensor"
sensor0     

lego-port port0: Added new device 'in2:ev3-uart-host'
00:18:42.818 [info] lego-port port0: Added new device 'in2:ev3-uart-host'

iex(andy@nerves-5e58)59> ls "/sys/class/lego-sensor"
sensor0
```     
sensor0 is a touch sensor, a non-UART sensor.

**Comment by fhunleth** [comment]
Could you try creating a rootfs_overlay directory in your project and turning off the debug console serial port? The file that you need to create is `/etc/modprobe.d/ev3dev.conf`. If it's an empty file, then it should work. Here are the default contents: https://github.com/nerves-project/nerves_system_ev3/blob/master/rootfs_overlay/etc/modprobe.d/ev3dev.conf. Back at the time, supporting the console uart made sense, but now there are more ways to avoid it. I'm wondering if the driver that's used for the console UART is also used for sensors and that's why it doesn't work for you.

**Comment by jfcloutier** [comment]
No dice.

I can open an ssh console on the EV3 so that part works great :-)
I emptied the `rootfs_overlay/etc/modprobe.d/ev3dev.conf` file, disabling the serial port console.
I plugged an EV3 touch sensor (non-UART) into port 1 and it appears under /sys/class/lego-sensor.
I plugger an EV3 color sensor (UART) into port 2 and it still does not appear under /sys/class/lego-sensor.

Adding (non-UART) touch sensor in port 1:
```
iex(andy@nerves-5e58)21> :os.cmd 'dmesg | tail'     
...Added new device \'in2:ev3-uart-host\'\nlego-sensor sensor1: Registered \'lego-ev3-touch\' on \'in1\'.\nlego-port port0: Added new device \'in1:lego-ev3-touch\'\n'
iex(andy@nerves-5e58)22> ls "/sys/class/lego-sensor"
sensor1     
```
Adding (UART) color sensor in port 2:
```
iex(andy@nerves-5e58)23> :os.cmd 'dmesg | tail'     
'...\nlego-port port1: Added new device \'in2:ev3-uart-host\'
iex(andy@nerves-5e58)24> ls "/sys/class/lego-sensor"
sensor1     
```
Unfortunately, disabling the serial port console did not fix the problem.

**Comment by jfcloutier** [comment]
The `Registered...` message does not appear in the dmesg log when plugging UART sensors.

**Comment by jfcloutier** [comment]
@fhunleth Do you still have the Ev3 brick and a UART sensor? Would it be possible to quickly see if you can duplicate the problem on your end? Maybe I'm doing something wrong but I can't see what it would be.

**Comment by fhunleth** [comment]
Yep - I have it. I'll make some time this week to try it out.

**Comment by jfcloutier** [comment]
Thanks! I appreciate.

**Comment by fhunleth** [comment]
It turned out that `ldattach` was no longer being built due to a change in `util-linux`. This will be fixed by #29.

**Comment by jfcloutier** [comment]
UART sensors are detected once again as of v1.3.0. Thanks!
